### PR TITLE
Fix more linting and formatting errors

### PR DESCRIPTION
**Pull request by lukebennett88**

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -4,7 +4,8 @@ const ERROR = 2;
 
 const reactComponentTypeMessage = {
   message:
-    'This type includes the children prop which is generally wrong, instead of using this type, type the props of the component',
+    'This type includes the children prop which is generally wrong, ' +
+    'instead of using this type, type the props of the component',
```
Comment (lukebennett88) [inline .eslintrc.js:8]: Splitting up the comments in this file so they're under 80 characters and easier to read

File: packages/text-input/src/TextInput.tsx [hunk 2]
```diff
@@ -60,13 +60,14 @@ export const TextInput = forwardRef<HTMLInputElement, TextInputProps>(
...
 );
+TextInput.displayName = 'TextInput';
 
 // Styled components
 // ------------------------------
 
 export type UseInputProps = Pick<FieldContextType, 'disabled' | 'invalid'>;
 
-export function useInput({ disabled, invalid }: UseInputProps) {
+export function useInput({ disabled }: UseInputProps) {
```
Comment (lukebennett88) [inline packages/text-input/src/TextInput.tsx:70]: I've removed the unused value here, but `invalid` is still required for `useInput`.
I did this for two reasons:
- Don't need to update every use of `useInput` to remove `invalid`
- We don't have any invalid styles yet, but we need to revisit them (e.g. if you focus on an input that is invalid, you get a green focus ring which looks wrong to me)

File: packages/text/src/defaultTextProps.tsx [hunk 3]
```diff
@@ -16,12 +15,12 @@ const DefaultTextPropsContext = createContext<DefaultTextProps>({
...
 });
 
-export const DefaultTextPropsProvider: React.FC<DefaultTextProps> = ({
+export function DefaultTextPropsProvider({
   children,
   size,
   tone,
   weight,
-}) => {
+}: DefaultTextProps & { children: ReactNode }) {
```
Comment (nderkim) [inline packages/text/src/defaultTextProps.tsx:23]: should we use `React.PropsWithChildren`?

Comment (lukebennett88) [reply packages/text/src/defaultTextProps.tsx:23]: `PropsWithChildren` has children as optional which it's not in this case

File: .eslintrc.js [hunk 4]
```diff
@@ -118,10 +128,27 @@ module.exports = {
...
       rules: {
-        // TypeScript already checks for the following things and they conflict with TypeScript
+        // TypeScript already checks for the following things and they cause conflicts
         'import/no-unresolved': OFF,
         'no-undef': OFF,
       },
     },
+    {
+      files: [
+        '**/__tests__/**/*',
```
Comment (freemany) [inline .eslintrc.js:138]: we don't have __tests__ folder

Comment (lukebennett88) [reply .eslintrc.js:138]: We don't right now, but might in the future.

File: packages/utils/src/useComposedRefs.ts [hunk 5]
```diff
@@ -38,9 +38,12 @@ export function assignRef<RefValueType = any>(
...
-      assignRef(ref, node);
-    }
-  }, refs);
+  return useCallback(
+    (node: any) => {
+      for (const ref of refs) {
+        assignRef(ref, node);
+      }
+    },
+    [refs]
```
[the remark this answers is not in the document]
Comment (lukebennett88) [reply packages/utils/src/useComposedRefs.ts:47]: Yeah I'm not 100% sure on this, starting to think it's not. Maybe we should override the ESLint rule here.

Or maybe we could use [this instead](https://github.com/Andarist/use-composed-ref/blob/main/src/index.ts)?

**Review by nderkim — APPROVED** [review]

**Pull request merged**